### DISCUSSION TITLE
End the shell when status of the last send is != 0 ?

### DISCUSSION THREAD

**Discussion by jipen** [JacobCallahan/Hussh#17] [Ideas]
Hello Jacob,

Perhaps add (if possible) a (boolean) parameter to the send function to stop the shell when the return status of the last command is not zero and set the shell.result.status to this value ?

Best regards.

**Comment by JacobCallahan** [comment]
Much like my answer in the other discussion here [1], this currently isn't possible because we'd need to close the channel before we can get the latest exit status. I'll keep both of these open in case someone comes up with a way to do this.

[1] - https://github.com/JacobCallahan/Hussh/discussions/18#discussioncomment-10835572

**Reply by jipen** [reply]
The only way that I found is what I do with Fabric:, instead of using a shell, grouping commands separated by "&&" in a string and then execute/run this string.
But then the result (stdout/stderr) can be a mess if there's too many concatenated commands.
It seems that a "real" SSH interactive shell where the context (working directory, env vars,...) is kept while executing each commands is a challenge as the others SSH libraries I know, such as Paramiko or Fabric, can't achieve it too because of this cursed channel which has to be closed to recover the status and the sdtout/stderr !

**Reply by JacobCallahan** [reply]
It's definitely a challenge, for sure. It is likely just a limitation of libssh2, which is the foundation of this library as well as paramiko/fabric, and of course ssh2-python. When I get a chance to work on the async connection class, I'll make sure to experiment with an async interactive shell.
